src/util/execute-lifecycle-script.js

```diff
@@ -142,6 +142,10 @@ export async function makeEnv(
   const envPath = env[constants.ENV_PATH_KEY];
   const pathParts = envPath ? envPath.split(path.delimiter) : [];
 
+  // Include the directory that contains node so that we can guarantee that the scripts
+  // will always run with the exact same Node release than the one use to run Yarn
+  pathParts.unshift(path.dirname(process.execPath));
```

Comment:
We should probably assign `path.dirname(process.execPath)` to a variable at this point since it is calculated over and over below.

Reply:
Yep, but I already plan to refactor this part in a followup!